kube-apiserver: Remove deprecated default service cluster IP support

masap opened the pull request:
**What type of PR is this?**
/kind deprecation

**What this PR does / why we need it**:
```TestGetServiceIPAndRanges``` in Unit test is testing ```getServiceIPAndRanges() ```and prints the following warning:

```
W1015 07:41:49.639419   98549 services.go:37] No CIDR for service cluster IPs specified. Default value which was 10.0.0.0/24 is deprecated and will be removed in future releases. Please specify it using --service-cluster-ip-range on kube-apiserver.
```
As this warning says, the default service cluster IP is deprecated. So this PR modifies ```getServiceIPAndRanges()``` not to use this default. Due to the change, if ```getServiceIPAndRanges()``` receives a ```""``` argument, an error will occur.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Another fix is ​​to ```getServiceIPAndRanges()``` return the default service cluster IP even after the default service cluster IP in ```controlplane.ServiceIPRange()``` is removed. I've fixed it this PR way for consistency, but any comments on this are appreciated!


**Does this PR introduce a user-facing change?**:
```release-note
The default value of 10.0.0.0/24 for the API server startup flag --service-cluster-ip-range is deprecated, so user need to specify some value.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

fedebongio commented:
/sig network
/assign @lavalamp 
(sorry!)
/triage accepted

masap commented:
@lavalamp Could you take a look this PR as well ?

aojea commented:
So, if I understand it correctly, right now the apiserver can work without any service-subnet, it just falls back to the 10.0.0.0/24 for the service subnet.
With this change the apiserver will not start and fail ... that seems like a tricky change, let´s ping @thockin for thoughts on this too ... my main concern is if there is people relying on this behaviour, this will break them

masap committed:
kube-apiserver: Remove default service cluster IP support

TestGetServiceIPAndRanges in Unit test is testing getServiceIPAndRanges () and
prints the following warning:

W1015 07:41:49.639419   98549 services.go:37] No CIDR for service cluster IPs specified. Default value which was 10.0.0.0/24 is deprecated and will be removed in future releases. Please specify it using --service-cluster-ip-range on kube-apiserver.

As this warning says, the default service cluster IP is deprecated. So this PR
modifies getServiceIPAndRanges () not to use this default. Due to the change,
if getServiceIPAndRanges () receives a "" argument, an error will occur.

masap commented:
rebased

masap commented:
@aojea  Thanks !

> So, if I understand it correctly, right now the apiserver can work without any service-subnet,
> it just falls back to the 10.0.0.0/24 for the service subnet.

Yes, trailing routine makes it fall back to the ```10.0.0.0/24``` when service cluster IP range is not specified.
But it is deprecated and will be removed in future releases as it says.
https://github.com/kubernetes/kubernetes/blob/9f2f1b8c460b4b5ec8bfcd34490fb259ebad65b0/pkg/controlplane/services.go#L36-L39

> With this change the apiserver will not start and fail ...

I can boot apiserver with ```local-up-cluster.sh```.
In addition, this PR passes all the integration tests with latest source codes.

masap commented:
@khenidak Fixed.

masap commented:
/ok-to-test

masap commented:
/retest

masap commented:
/test pull-kubernetes-e2e-kind

masap commented:
/test pull-kubernetes-unit

masap commented:
/test pull-kubernetes-node-e2e-containerd

khenidak commented:
/assign @khenidak 

This will break a lot of bootstrap tools. Most of the tools just depend on api-server to do the right thing and default service CIDR (which instead we want them to set it).

Do we really want to remove that band aid now?

masap commented:
If the default is used still now, should we remove the warning message ?
```Default value which was 10.0.0.0/24 is deprecated and will be removed in future releases.```

aojea commented:
> If the default is used still now, should we remove the warning message ?
> `Default value which was 10.0.0.0/24 is deprecated and will be removed in future releases.`

I think that we need to live with this ... nobody should depend on that, but we can't remove it because it can break someone

masap commented:
Ah, OK ! I agree. I will drop this PR.
